src/requirements/requirement-fronend-computation.ts

```diff
@@ -123,6 +128,45 @@ const getTotalCreditsFulfillmentStatistics = (
   };
 };
 
+const getSwimTestFulfillmentStatistics = (
+  college: string,
+  courses: readonly CourseTaken[],
+  tookSwimTest: boolean
+): FulfillmentStatistics => {
+  const requirement: RequirementWithIDSourceType = {
+    id: 'College-UNI-SwimTest',
+    sourceType: 'College',
+    sourceSpecificName: college,
+    name: 'Swim Test',
+    description:
+      'The Faculty Advisory Committee on Athletics and Physical Education has established a basic swimming ' +
+      'and water safety competency requirement for all entering first-year undergraduate students.',
+    source: 'http://courses.cornell.edu/content.php?catoid=41&navoid=11637',
+    courses: [Object.fromEntries(ALL_ROSTERS.map(roster => [roster, [SWIM_TEST_COURSE_ID]]))],
+    subRequirementProgress: 'any-can-count',
+    fulfilledBy: 'courses',
+    minCount: 1,
+  };
+  const swimClasses = courses.filter(it => it.courseId === SWIM_TEST_COURSE_ID);
+  if (tookSwimTest) {
+    swimClasses.push({
+      roster: 'SP21',
```

Comment:
To confirm: Would we want the roster field value to be the most recent roster we have course data for?

Reply:
updated to use generated roster json